kani-compiler/src/codegen_cprover_gotoc/reachability.rs

```diff
@@ -457,67 +456,97 @@ fn should_codegen_locally<'tcx>(tcx: TyCtxt<'tcx>, instance: &Instance<'tcx>) ->
     }
 }
 
-/// Extract the pair (concrete, trait) for a unsized cast.
-/// This function will return None if this is not a unsizing coercion from concrete to trait.
+/// Extract the pair (from_ty, to_ty) for a unsized cast.
 ///
 /// For example, if `&u8` is being converted to `&dyn Debug`, this method would return:
-/// `Some(u8, dyn Debug)`.
+/// `(u8, dyn Debug)`.
 ///
 /// This method also handles nested cases and `std` smart pointers. E.g.:
 ///
 /// Conversion between `Rc<Wrapper<String>>` into `Rc<Wrapper<dyn Debug>>` should return:
```

Comment:
Some more context about how this can happen would be appreciated here. I don't think Rust lets you write this kind of conversion, so is this only something that can happen in the context of a dyn-ptr to dyn-ptr cast?

Basically, when does a "coerce unsized" even happen?

Reply:
Not sure I understand what you mean by "I don't think Rust let's you write this type of conversion."

But I will try to improve the doc here.